src/app/modules/main/controller.nim

```diff
@@ -278,6 +282,14 @@ proc init*(self: Controller) =
   self.events.on(SIGNAL_COMMUNITY_MY_REQUEST_ADDED) do(e: Args):
     self.delegate.onMyRequestAdded();
 
+  self.events.on(SIGNAL_COMMUNITY_TOKEN_DEPLOYED) do(e: Args):
+    let args = CommunityTokenDeployedArgs(e)
+    self.delegate.onCommunityTokenDeployed(args.communityToken)
+
+  self.events.on(SIGNAL_COMMUNITY_TOKEN_DEPLOY_STATUS) do(e: Args):
+    let args = CommunityTokenDeployedStatusArgs(e)
+    self.delegate.onCommunityTokenDeployStateChanged(args.communityId, args.contractAddress, args.deployState)
```

Comment:
Those should not be listened in the main controller. You can put it in the section controller and put an `if` to check if `args.communityId` is equal to the `sectionId`. That way, the main module doesn't have to loop sections. The personal chat section also doesn't have to listen to it (we have a check for `isCommunity` or something already in the section controller).

Reply:
Ok, turns out I misunderstood this code. The main module has to do it, because it needs to change the section **item**, not the section **module**, and it's way simpler for main to do it, because it is the parent of section_item

Reply:
We discussed this. In the future we should have some interaction between section_item and corresponding module.